src/d3d9/d3d9_common_buffer.cpp

```diff
@@ -74,14 +74,6 @@ namespace dxvk {
     if (!(m_desc.Usage & (D3DUSAGE_DYNAMIC | D3DUSAGE_WRITEONLY)))
       return D3D9_COMMON_BUFFER_MAP_MODE_BUFFER;
 
-    // Tests show that DISCARD does not work for pure SWVP devices.
-    // So force staging buffer path to avoid stalls.
-    // Dark Romance: Vampire in Love also expects draws to be synchronous
-    // and breaks if we respect NOOVERWRITE.
-    // D&D Temple of Elemental Evil breaks if we respect DISCARD. 
-    if (m_parent->CanOnlySWVP())
```

Comment:
Have we tested this on `D&D Temple of Elemental Evil` that it is actually this?

Reply:
Current version of the PR is fine with ToEE (at some point it broke it, but was corrected).